test/fft_test.rg

```diff
@@ -1,108 +1,409 @@
+task print_array_double_complex(input : region(ispace(int1d), complex64), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  for x in input do
+    var currComplex = input[x]
+    format.println("{} + {}j, ", currComplex.real, currComplex.imag)
+  end
+  format.println("]\n")
+end
+
+-- Task to print out input or output array with fieldspace complex32. Takes a region and a string representing the name of the array
+__demand(__inline, __leaf)
+task print_array_float_complex(input : region(ispace(int1d), complex32), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  for x in input do
+    var currComplex = input[x]
+    format.println("{} + {}j, ", currComplex.real, currComplex.imag)
+  end
+  format.println("]\n")
+end
+
+-- Task to print out input or output array with fieldspace double
+__demand(__inline, __leaf)
+task print_array_double_real(input : region(ispace(int1d), double), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  for x in input do
+    var currInt = input[x]
+    format.println("{}, ", currInt)
+  end
+  format.println("]\n")
+end
+
+
+-- Task to print out input or output array with fieldspace float
+__demand(__inline, __leaf)
+task print_array_float_real(input : region(ispace(int1d), float), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  for x in input do
+    var currInt = input[x]
+    format.println("{}, ", currInt)
+  end
+  format.println("]\n")
+end
+
+
+-- Task to print out input or output 2D array with fieldspace complex64
+task print_array_2d_double_complex(input : region(ispace(int2d), complex64), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  format.println("Bounds = {}", input.bounds)
+  for x in input do
+    var currComplex = input[x]
+    format.println("index {}: {} + {}j, ", x, currComplex.real, currComplex.imag)
+  end
+  format.println("]\n")
+end
+
+-- Task to print out input or output 3D array with fieldspace complex64
+task print_array_3d_double_complex(input : region(ispace(int3d), complex64), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  format.println("Bounds = {}", input.bounds)
+  for x in input do
+    var currComplex = input[x]
+    format.println("index {}: {} + {}j, ", x, currComplex.real, currComplex.imag)
+  end
+  format.println("]\n")
+end
+
+-- Task to print out input or output 3D array with fieldspace double
+task print_array_3d_double_real(input : region(ispace(int3d), double), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  format.println("Bounds = {}", input.bounds)
+  for x in input do
+    var currInt = input[x]
+    format.println("{}, ", currInt)
+  end
+  format.println("]\n")
+end
+
+--task compare_regions_double_complex(r1 : region(ispace(int1d), complex64), r2 : region(ispace(int1d), complex64))
```

Comment:
We don't actually test anything?

Reply:
Removed this. Have a `make_compare_regions_task` through metaprogramming now.